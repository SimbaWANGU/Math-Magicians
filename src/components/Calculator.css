* {
  width: 100%;
  height: 100%;
}

.calculatorContainer {
  border-top: 1px solid black;
  border-left: 1px solid black;
  border-right: 1px solid black;
  margin: auto;
  width: 50%;
  height: 70%;
}

.resultContainer {
  width: 100%;
  padding: 0;
}

.firstRow {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 16%;
  text-align: center;
}

.secondRow {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 16%;
  text-align: center;
}

.thirdRow {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 16%;
  text-align: center;
}

.fourthRow {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 16%;
  text-align: center;
}

.fifthRow {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 15.5%;
  text-align: center;
  border-bottom: 1px solid black;
}

.lastRow {
  display: flex;
  flex-direction: row;
  width: 100%;
}
