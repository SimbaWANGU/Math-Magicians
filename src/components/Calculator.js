import React from 'react';
import './Calculator.css';
import Result from './Result';
import InputValue from './InputValue';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      obj: {
        total: null,
        next: null,
        operation: null,
      },
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(btnValue) {
    const { obj } = this.state;
    const newObj = { ...obj, ...calculate(obj, btnValue) };
    this.setState({
      obj: newObj,
    });
  }

  render() {
    const { obj } = this.state;
    return (
      <div className="calculatorContainer">
        <div>
          <Result result={obj.next || obj.total || 0} className="resultContainer" />
          <div className="firstRow">
            <InputValue content="AC" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="+/-" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="%" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="รท" backgroundColor="Moccasin" handleClick={this.handleClick} />
          </div>
          <div className="secondRow">
            <InputValue content="7" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="8" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="9" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="x" backgroundColor="Moccasin" handleClick={this.handleClick} />
          </div>
          <div className="thirdRow">
            <InputValue content="4" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="5" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="6" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="-" backgroundColor="Moccasin" handleClick={this.handleClick} />
          </div>
          <div className="fourthRow">
            <InputValue content="1" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="2" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="3" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <InputValue content="+" backgroundColor="Moccasin" handleClick={this.handleClick} />
          </div>
          <div className="fifthRow">
            <InputValue content="0" backgroundColor="GhostWhite" handleClick={this.handleClick} />
            <div className="lastRow">
              <InputValue content="." backgroundColor="GhostWhite" handleClick={this.handleClick} />
              <InputValue content="=" backgroundColor="Moccasin" handleClick={this.handleClick} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
