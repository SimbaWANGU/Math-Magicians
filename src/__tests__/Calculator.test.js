import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../components/Calculator';

describe('Calculator tests', () => {
  it('Calculator renders as expected', () => {
    const component = renderer.create(
      <Calculator />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Calculator is visible to the user', () => {
    const { getByTestId } = render(<Calculator />);
    expect(getByTestId('caculatorDisplay')).toBeInTheDocument();
  });
});
