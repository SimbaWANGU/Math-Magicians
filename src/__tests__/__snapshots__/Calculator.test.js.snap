// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator tests Calculator renders as expected 1`] = `
<div
  className="calculatorContainer"
  data-testid="caculatorDisplay"
>
  <div>
    <input
      className="resultContainer"
      data-testid="resultDisplay"
      disabled={true}
      onKeyDown={[Function]}
      type="text"
      value={0}
    />
    <div
      className="firstRow"
    >
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          AC
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          +/-
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          %
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "Moccasin",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          รท
        </p>
      </div>
    </div>
    <div
      className="secondRow"
    >
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          7
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          8
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          9
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "Moccasin",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          x
        </p>
      </div>
    </div>
    <div
      className="thirdRow"
    >
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          4
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          5
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          6
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "Moccasin",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          -
        </p>
      </div>
    </div>
    <div
      className="fourthRow"
    >
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          1
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          2
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          3
        </p>
      </div>
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "Moccasin",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          +
        </p>
      </div>
    </div>
    <div
      className="fifthRow"
    >
      <div
        className="inputValue"
        style={
          Object {
            "backgroundColor": "GhostWhite",
          }
        }
      >
        <p
          className="nameOfValue"
          data-testid="calcValue"
          onClick={[Function]}
        >
          0
        </p>
      </div>
      <div
        className="lastRow"
      >
        <div
          className="inputValue"
          style={
            Object {
              "backgroundColor": "GhostWhite",
            }
          }
        >
          <p
            className="nameOfValue"
            data-testid="calcValue"
            onClick={[Function]}
          >
            .
          </p>
        </div>
        <div
          className="inputValue"
          style={
            Object {
              "backgroundColor": "Moccasin",
            }
          }
        >
          <p
            className="nameOfValue"
            data-testid="calcValue"
            onClick={[Function]}
          >
            =
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
